### DISCUSSION TITLE
How come the closure for addColumns does NOT pass the row index?

### DISCUSSION THREAD

**Discussion by thedangler** [Power-Components/livewire-powergrid#752] [Q&A]
Wondering why when trying to  manage columns you can't access the index of the record you are looping?

For example if doing a custom field
->addColumn('myCustomField', function($col){
  // how do I get the current iteration of row?
});

// this would be nice. Unless there is a way to do it I can't find in the docs
->addColumn('myCustomField', function($col, $index){
  // how do I get the current iteration of row?
});

Please not this is for a Collection.

**Comment by prasetyodidi** [comment]
you can make a private variable say it "index" with default value is 1. on your return closure function $this->index++;
